addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -1,40 +1,68 @@
 odoo.define("pos_gift_card.GiftCardPopup", function (require) {
   "use strict";
 
-  const { useState, useRef } = owl.hooks;
+  const { useState } = owl.hooks;
   const AbstractAwaitablePopup = require("point_of_sale.AbstractAwaitablePopup");
+  const { useBarcodeReader } = require("point_of_sale.custom_hooks");
   const Registries = require("point_of_sale.Registries");
 
   class GiftCardPopup extends AbstractAwaitablePopup {
     constructor() {
       super(...arguments);
+
+      this.confirmFunctions = {
+        'create_set': this.generateBarcode.bind(this),
+        'scan_set': this.scanAndUseGiftCard.bind(this),
+        'scan_use': this.scanAndUseGiftCard.bind(this),
+        'pay': this.payWithGiftCard.bind(this),
+        'showAmount': this.ShowRemainingAmount.bind(this)
+      };
+
       this.state = useState({
         giftCardConfig: this.env.pos.config.gift_card_settings,
-        showBarcodeGeneration: false,
-        showNewGiftCardMenu: false,
-        showUseGiftCardMenu: false,
-        showGiftCardDetails: false,
+        showMenu: true,
+        error: '',
+        context: '',
```

Comment:
Instead of `context`, can you replace it with `command`? What do you think @trj-odoo?